File: app/src/main/java/com/nolamarel/onlinelibrary/Fragments/SupportServiceFragment.java
package com.nolamarel.onlinelibrary.Fragments;

import android.os.Bundle;

import androidx.fragment.app.Fragment;

import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.Toast;

import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.database.FirebaseDatabase;
import com.nolamarel.onlinelibrary.databinding.FragmentSupportServiceBinding;

import java.util.HashMap;

public class SupportServiceFragment extends Fragment {
    private FragmentSupportServiceBinding binding;
    private String problemText;
    private String emailText;
    private String messageText;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {

        binding = FragmentSupportServiceBinding.inflate(inflater, container, false);

        String userId = FirebaseAuth.getInstance().getCurrentUser().getUid().toString();

        binding.arrowBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                getParentFragmentManager().popBackStack();
            }
        });

        binding.supSerMessageEt.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                int characterCount = s.length();
                binding.supSerSymbolNow.setText(String.valueOf(characterCount) + "/20000");
            }
            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        EditText problem = binding.supSerThemeEt;
        EditText email = binding.supSerEmailEt;
        EditText message = binding.supSerMessageEt;

        binding.button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                boolean allFieldsValid = true;

                problemText = binding.supSerThemeEt.getText().toString();
                emailText = binding.supSerEmailEt.getText().toString();
                messageText = binding.supSerMessageEt.getText().toString();

                if (problemText.isEmpty()){
                    problem.setError("Введите проблему");
                    allFieldsValid = false;
                }
                if (emailText.isEmpty()){
                    email.setError("Введите адрес электронной почты");
                    allFieldsValid = false;
                }
                if (messageText.length() < 20){
                    message.setError("Слишком короткое описание");
                    allFieldsValid = false;
                }

                if (allFieldsValid) {
                    sendMessage(userId, problemText, emailText, messageText);
                    Toast.makeText(getContext(), "Сообщение успешно отправлено", Toast.LENGTH_SHORT).show();
                }
            }
        });




        return binding.getRoot();
    }

    public void sendMessage(String userId, String problemText, String emailText, String messageText){
        if (userId == null) return;
        HashMap<String, String> messageInfo = new HashMap<>();
        messageInfo.put("userId", userId);
        messageInfo.put("problem", problemText);
        messageInfo.put("email", emailText);
        messageInfo.put("message", messageText);
        FirebaseDatabase.getInstance().getReference().child("Messages").push().setValue(messageInfo);
    }

}

File: app/src/main/java/com/nolamarel/onlinelibrary/Fragments/SupportServiceFragment.kt
package com.nolamarel.onlinelibrary.Fragments

import android.content.Intent
import android.net.Uri
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.util.Patterns
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import com.nolamarel.onlinelibrary.databinding.FragmentSupportServiceBinding

class SupportServiceFragment : Fragment() {

    private var _binding: FragmentSupportServiceBinding? = null
    private val binding get() = _binding!!

    private val supportEmail = "support@onlinelibrary.local"

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentSupportServiceBinding.inflate(inflater, container, false)

        binding.arrowBack.setOnClickListener {
            parentFragmentManager.popBackStack()
        }

        binding.supSerMessageEt.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) = Unit

            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
                val characterCount = s?.length ?: 0
                binding.supSerSymbolNow.text = "$characterCount/20000"
            }

            override fun afterTextChanged(s: Editable?) = Unit
        })

        binding.button.setOnClickListener {
            validateAndSend()
        }

        return binding.root
    }

    private fun validateAndSend() {
        val problem = binding.supSerThemeEt.text.toString().trim()
        val email = binding.supSerEmailEt.text.toString().trim()
        val message = binding.supSerMessageEt.text.toString().trim()

        clearErrors()

        var isValid = true

        if (problem.isEmpty()) {
            binding.supSerThemeEt.error = "Введите тему обращения"
            isValid = false
        }

        if (email.isEmpty()) {
            binding.supSerEmailEt.error = "Введите email"
            isValid = false
        } else if (!Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            binding.supSerEmailEt.error = "Введите корректный email"
            isValid = false
        }

        if (message.length < 20) {
            binding.supSerMessageEt.error = "Слишком короткое описание"
            isValid = false
        }

        if (!isValid) return

        sendSupportEmail(
            problemText = problem,
            emailText = email,
            messageText = message
        )
    }

    private fun clearErrors() {
        binding.supSerThemeEt.error = null
        binding.supSerEmailEt.error = null
        binding.supSerMessageEt.error = null
    }

    private fun sendSupportEmail(
        problemText: String,
        emailText: String,
        messageText: String
    ) {
        val subject = "Обращение в поддержку: $problemText"
        val body = buildString {
            appendLine("Тема: $problemText")
            appendLine("Email для ответа: $emailText")
            appendLine()
            appendLine("Сообщение:")
            appendLine(messageText)
        }

        val intent = Intent(Intent.ACTION_SENDTO).apply {
            data = Uri.parse("mailto:$supportEmail")
            putExtra(Intent.EXTRA_SUBJECT, subject)
            putExtra(Intent.EXTRA_TEXT, body)
        }

        if (intent.resolveActivity(requireActivity().packageManager) != null) {
            startActivity(intent)
        } else {
            Toast.makeText(
                requireContext(),
                "На устройстве нет почтового приложения",
                Toast.LENGTH_LONG
            ).show()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}